custom/icds_reports/data_pull/exporter.py

```diff
@@ -30,11 +31,16 @@ def __init__(self, slug_or_file, db_alias, month, location_id):
     @prevent_parallel_execution(DATA_PULL_CACHE_KEY)
     def export(self):
         zip_file_name = "%s-DataPull.zip" % self.data_pull_obj.name
-        with zipfile.ZipFile(zip_file_name, mode='a') as z:
-            for filename, string_buffer in self.data_pull_obj.run().items():
-                z.writestr(filename, string_buffer.getvalue())
+        with zipfile.ZipFile(zip_file_name, mode='w') as z:
+            for filename, data in self.data_pull_obj.run().items():
+                z.writestr(filename, self.get_file_content(data))
```

Comment:
you can also avoid overriting this once you have xlsx as filestream.

Reply:
removed